File: bind/pdflib/perl/starter_tagged.pl
#!/usr/bin/perl
# $Id: starter_tagged.pl,v 1.3.2.2 2008/12/25 13:28:41 rp Exp $
#
# Tagged PDF starter:
# Create document with structure information for reflow and accessibility
#
# required software: PDFlib/PDFlib+PDI/PPS 7
# required data: none (dummy text created in program)

use pdflib_pl 7.0;

# This is where the data files are. Adjust as necessary.
$searchpath = "../data";
$outfilename = "starter_tagged.pdf";

$p = PDF_new();

eval {
    # This means we must check return values of load_font() etc.
    PDF_set_parameter($p, "errorpolicy", "return");

    PDF_set_parameter($p, "SearchPath", $searchpath);

    if (PDF_begin_document($p, $outfilename, "tagged=true") == -1) {
	die("Error: " . PDF_get_errmsg($p));
    }

    PDF_set_info($p, "Creator", "PDFlib starter sample");
    PDF_set_info($p, "Title", "starter_tagged");

    # Automatically create spaces between chunks of text
    PDF_set_parameter($p, "autospace", "true");

    # open the first structure element as a child of the document
    # structure root (=0)
    
    $id = PDF_begin_item($p, "P", "Title = {Simple Paragraph}");

    PDF_begin_page_ext($p, 0, 0, "width=a4.width height=a4.height");

    $font = PDF_load_font($p, "Helvetica", "winansi", "");
    if ($font == -1) {
	die("Error: %s\n", PDF_get_errmsg($p));
    }
    PDF_setfont($p, $font, 24.0);

    PDF_show_xy($p, "Hello, Tagged PDF!", 50, 700);
    PDF_continue_text($p, "This PDF has a very simple");
    PDF_continue_text($p, "document structure.");

    PDF_end_item($p, $id);

    # The page number is created as an artifact; it will be
    # ignored when reflowing the page in Acrobat.

    $id_artifact = PDF_begin_item($p, "Artifact", "");
    PDF_show_xy($p, "Page 1", 250, 100);
    PDF_end_item($p, $id_artifact);

    PDF_end_page_ext($p, "");

    PDF_end_document($p, "");

    PDF_delete($p);
};

if ($@) {
    printf("starter_tagged: PDFlib Exception occurred:\n");
    printf(" $@\n");
    exit;
}

File: bind/pdflib/php/examples.php5/starter_tagged.php
<?php
/* $Id: starter_tagged.php,v 1.1.2.2 2009/08/07 07:30:46 rjs Exp $
 *
 * Tagged PDF starter:
 * Create document with structure information for reflow and accessibility
 *
 * required software: PDFlib/PDFlib+PDI/PPS 7
 * required data: none (dummy text created in program)
 */

/* This is where the data files are. Adjust as necessary. */
$searchpath = "../data";
$outfilename = "starter_tagged.pdf";

try {
    $p = new PDFlib();

    # This means we must check return values of load_font() etc.
    $p->set_parameter("errorpolicy", "return");

    $p->set_parameter("SearchPath", $searchpath);

    /* "lang" contains the predominant document language */
    if ($p->begin_document($outfilename, "tagged=true lang=en") == 0) {
	die("Error: " . $p->get_errmsg());
    }

    $p->set_info("Creator", "PDFlib starter sample");
    $p->set_info("Title", "starter_tagged");

    /* Automatically create spaces between chunks of text */
    $p->set_parameter("autospace", "true");

    /* open the first structure element as a child of the document
     * structure root (=0)
     */
    $id = $p->begin_item("P", "Title = {Simple Paragraph}");

    $p->begin_page_ext(0, 0, "width=a4.width height=a4.height");

    $font = $p->load_font("Helvetica", "winansi", "");
    if ($font == 0) {
	die("Error: " . $p->get_errmsg());
    }
    $p->setfont($font, 24.0);

    $p->show_xy("Hello, Tagged PDF!", 50, 700);
    $p->continue_text("This PDF has a very simple");
    $p->continue_text("document structure.");

    $p->end_item($id);

    /* The page number is created as an artifact; it will be
     * ignored when reflowing the page in Acrobat.
     */
    $id_artifact = $p->begin_item("Artifact", "");
    $p->show_xy("Page 1", 250, 100);
    $p->end_item($id_artifact);

    $p->end_page_ext("");

    $p->end_document("");

    print "$outfilename generated";

}
catch (PDFlibException $e) {
    die("PDFlib exception occurred in starter_tagged sample:\n" .
	"[" . $e->get_errnum() . "] " . $e->get_apiname() . ": " .
	$e->get_errmsg() . "\n");
}
catch (Exception $e) {
    die($e);
}

$p = 0;
?>
